src/rgw/rgw_main.cc

```diff
@@ -403,10 +403,26 @@ int main(int argc, const char **argv)
 
       fe = new RGWFCGXFrontend(fcgi_pe, config);
     } else if (framework == "civetweb" || framework == "mongoose") {
-      int port;
-      config->get_val("port", 80, &port);
+      string port;
+      int portno;
+      char *portend;
+      config->get_val("port", "80", &port);
+      const char *portstart = port.c_str();
+      portno = strtol(portstart, &portend, 10);
+      if (portstart == portend) {
+	derr << "ERROR: cannot parse port " << port << dendl;
+	return -1;
+      }
+      if (*portend && (*portend != 's' || portend[1])) {
+	derr << "ERROR: trailing stuff on port " << port << dendl;
+	return -1;
+      }
+      if (portno <= 0 || portno > 65535) {
+	derr << "ERROR: port out of range " << port << dendl;
+	return -1;
+      }
```

Comment:
Well, actually, civetweb doesn't start up if you give it a bad port.  It fails in a much more confusing way later on.

However, you are right, it's possible to pass an ip:port in.  Or [ipv6]:port.  And append one of [sr,].  And it's possible to pass in non-zero bits >65535 in portno and have them ignored.  The ssl redir feature isn't useful to us currently, since we don't yet support listening for both ssl|non-ssl.  And we'll never pass in a , because the config file parser eats those elsewhere.  I don't mind breaking rgw for people who set high order bits in portno.

But being able to specify a specific ipaddr is useful, sigh.